### DISCUSSION TITLE
想讨论下pkg的规范问题

### DISCUSSION THREAD

**Discussion by constly** [ejoy/ant#105] [General]
Ant是一个非常模块化的引擎，未来一定会出现各种 pkg包 互相分享复用的情况，
我感觉目前的包机制可以多一些规范，以便后面分享协作。
具体而言，我觉得包内有必要维护一个类似package.lua的文件，里面配置大约如下:
```
return 
{
	version = 0.1,					-- 包版本
	engine_version = 1.2,				-- 依赖的引擎版本
	desc = "",					-- 包功能描述
	author = "",					-- 作者
	address = "",					-- git 地址啥的
	dep = {						-- 依赖的包和版本
		{name = "ant.efk", version = 0.5},
		{name = "ant.asset", version = 1},
		{name = "pkg.other", version = 1},
    },
}
```
最主要的是：包版本，依赖的引擎版本，依赖的包情况（目前包内可以随意import其他包，感觉太随意了，根本没法一眼看出某个包的依赖项）
一般来讲，随着功能越来越复杂，依赖链也会越来越复杂，这个过程中最麻烦的，某些作者维护的包升级比较频繁，有些可能很久都不会更新一次，如果没有版本号之类的东西，感觉后面会很混乱。

大家怎么看呢？ （当然这个事情目前来讲优先级不是很高）

**Comment by cloudwu** [comment]
目前这个阶段，给 pkg 维护版本号还不太现实。不太可能新旧版本混用。得等稳定后才有可能做。

**Reply by constly** [reply]
希望未来能考虑 pkg依赖 也需要配置，这样方便有兴趣的人做包管理。
对于包管理，之前也有个这方面的讨论，其实我个人觉得这个包管理后面真的可以考虑做一下。
想象一下，如果有一个地方可以看到各种各样的功能/效果包，可以随时下载运行看看效果，这对引擎发展、对调动大家的积极性是非常有帮助的。
如果用github之类的，一是太分散了，根本不知道谁谁做了什么东西，不好站在前人的肩膀上更进一步；二是由于墙的原因，用起来始终不太方便。
